### PR TITLE
refactor: Replace Akka with Pekko

### PR DESCRIPTION
**Pull request by seakayone**
Apache Pekko is a fork of Akka 2.6.x, prior to the Akka project’s adoption of the Business Source License. Version 1.0.0 is a direct replacement of Akka 2.6.x and has support for Scala 3 already. https://pekko.apache.org/

See also the RFC for the proposed switch: https://www.notion.so/review-RFC-006-Replace-Akka-with-Pekko-12c2199a8bff43adb4d907073d33e878?pvs=4

### Task Description/Number

Issue Number: DEV-2709

### Basic Requirements

Please check if your PR fulfills the following requirements:

- [ ] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)

### PR Type

What kind of change does this PR introduce?

- [ ] fix: represents bug fixes
- [x] refactor: represents production code refactoring
- [ ] feat: represents a new feature
- [ ] docs: documentation changes (no production code change)
- [ ] chore: maintenance tasks (no production code change)
- [ ] test: all about tests: adding, refactoring tests (no production code change)
- [ ] other... Please describe:

### Does this PR introduce a breaking change?

- [ ] Yes
- [x] No
- [ ] Maybe (not 100% sure => check with FE)

### Does this PR change client-test-data?

- [ ] Yes (don't forget to update the JS-LIB team about the change)
- [x] No

### CODE REVIEW

File: integration/src/test/scala/org/knora/webapi/responders/v2/OntologyResponderV2Spec.scala
Comment (BalduinLandolt) [inline integration/src/test/scala/org/knora/webapi/responders/v2/OntologyResponderV2Spec.scala:1]: this seems like a lot of similar cases where the fully qualified name is used in identical situations. Could `pekko.actor.Status.Failure` simply be imported so we'd only have  `case msg: Failure =>` or is there some nasty name conflict preventing this?

File: mkdocs.yml [hunk 1]
```diff
@@ -52,15 +52,15 @@ nav:
...
               - 0003 Change IIIF Service Manager and Sipi implementation to zlayer: 05-internals/design/adr/ADR-0003-change-iiif-service-manager-and-sipi-implementation-to-zlayer.md
               - 0004 Change Triplestore Service Manager and Fuseki implementation to ZLayer: 05-internals/design/adr/ADR-0004-change-triplestore-service-manager-and-fuseki-implementation-to-zlayer.md
               - 0005 Change ResponderManager to a simple case class: 05-internals/design/adr/ADR-0005-change-respondermanager-to-a-simple-case-class.md
-              - 0006 Gradually Replace AKKA-HTTP with ZIO-HTTP: 05-internals/design/adr/ADR-0006-use-zio-http.md
+              - 0006 Gradually Replace pekko-HTTP with ZIO-HTTP: 05-internals/design/adr/ADR-0006-use-zio-http.md
               - 0007 ZIO-fication of Responders: 05-internals/design/adr/ADR-0007-zio-fication-of-responders.md
           - DSP-API Design Principles:
               - Design Overview: 05-internals/design/principles/design-overview.md
-              - Futures with Akka: 05-internals/design/principles/futures-with-akka.md
+              - Futures with pekko: 05-internals/design/principles/futures-with-pekko.md
```
Comment (BalduinLandolt) [inline mkdocs.yml:63]: in title case spelling, upper case `Pekka` (or `PEKKA`) would probably be more appropriate than lower case

File: webapi/src/main/scala/org/knora/webapi/routing/KnoraRoute.scala [hunk 2]
```diff
@@ -20,14 +16,20 @@ import scala.concurrent.Future
...
+
 /**
  * Data that needs to be passed to each route.
  *
  * @param system    the actor system.
  * @param appActor  the main application actor.
  * @param appConfig the application's configuration.
  */
-case class KnoraRouteData(system: akka.actor.ActorSystem, appActor: akka.actor.ActorRef, appConfig: AppConfig)
+case class KnoraRouteData(system: pekko.actor.ActorSystem, appActor: pekko.actor.ActorRef, appConfig: AppConfig)
```
Comment (BalduinLandolt) [inline webapi/src/main/scala/org/knora/webapi/routing/KnoraRoute.scala:32]:
```suggestion
case class KnoraRouteData(system: ActorSystem, appActor: ActorRef, appConfig: AppConfig)
```

should be sufficient, looking at the imports (L19-20 in new line count)

File: webapi/src/main/scala/org/knora/webapi/routing/admin/ProjectsRouteADM.scala [hunk 3]
```diff
@@ -37,9 +27,21 @@ import org.knora.webapi.routing.RouteUtilADM._
...
+import pekko.http.scaladsl.server.PathMatcher
+import pekko.http.scaladsl.server.RequestContext
+import pekko.http.scaladsl.server.Route
+import pekko.stream.IOResult
+import pekko.stream.scaladsl.FileIO
+
 final case class ProjectsRouteADM()(
   private implicit val runtime: Runtime[
-    Authenticator with StringFormatter with MessageRelay with ProjectADMRestService
+    org.knora.webapi.routing.Authenticator with StringFormatter with MessageRelay with ProjectADMRestService
```
Comment (BalduinLandolt) [inline webapi/src/main/scala/org/knora/webapi/routing/admin/ProjectsRouteADM.scala:44]: why this change?

**Review by BalduinLandolt — APPROVED** [review]
See my comment in the RFC: I'm happy to see this merged.

All my remarks are details concerning imports. Feel free to ignore them.

File: webapi/src/main/scala/org/knora/webapi/routing/admin/ProjectsRouteADM.scala (hunk 3 above)
Comment (seakayone) [reply webapi/src/main/scala/org/knora/webapi/routing/admin/ProjectsRouteADM.scala:44]: For some reason the compiler fails if it is not the fully qualified class name.

e.g.:
```
[error] /Users/christian/git/dasch/dsp-api/webapi/src/main/scala/org/knora/webapi/routing/admin/ProjectsRouteADM.scala:296:30: could not find implicit value for parameter r: zio.Runtime[org.knora.webapi.routing.Authenticator with Any with org.knora.webapi.slice.admin.api.service.ProjectADMRestService with Any]
[error]     UnsafeZioRun.runToFuture {
```

**Review by mpro7 — APPROVED** [review]
